pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py

```diff
@@ -18,16 +18,24 @@ def _VSCODE_convertNumpyArrayToDataFrame(ndarray):
     try:
         x_len = temp.shape[0]
         # Figure out if we're dealing with ragged data
-        # Handle ragged arrays by making a container where the number of
-        # columns is the max length of all rows. Missing elements are
-        # represented as empty strings.
         row_lengths = [
             len(temp[i])
             if isinstance(temp[i], (list, tuple, _VSCODE_np.ndarray))
             else 1
             for i in range(x_len)
         ]
-        flattened = _VSCODE_np.full((x_len, max(row_lengths)), "", dtype="object")
+        max_y_len = max(row_lengths)
+        is_ragged_or_ndimensional = hasattr(temp, "ndim") and (
+            (temp.ndim == 1 and max_y_len > 1) or temp.ndim > 2
+        )
+
+        if not is_ragged_or_ndimensional:
+            return _VSCODE_pd.DataFrame(temp)
+
+        # Handle ragged arrays by making a container where the number of
+        # columns is the max length of all rows. Missing elements are
+        # represented as empty strings.
+        flattened = _VSCODE_np.full((x_len, max_y_len), "", dtype="object")
```

Comment:
I think we prefix all variables with _VSCODE_ so we can make sure we don't delete existing variables?

Reply:
For example if in the kernel I had a variable named flattened, doesn't this delete it/change it?

Reply:
Based on https://docs.python.org/3/faq/programming.html#what-are-the-rules-for-local-and-global-variables-in-python, 

> If a variable is assigned a value anywhere within the function’s body, it’s assumed to be a local unless explicitly declared as global.

For example,

![image](https://user-images.githubusercontent.com/30305945/106972960-b9d2d980-6706-11eb-9020-080efc3f24f5.png)

That being said, I'm no Python expert...

Reply:
TIL I learned there locally scoped variables in python. I thought they were all global